### ISSUE TITLE
Failed deliveries are reported via mail but not visible in the web ui

### ISSUE THREAD

**Issue by croxxx** [anonaddy/docker#111]
Today I had a reply mail bounced from a receiver and noticed something strange.

## Description

A few minutes after sending the mail, I received a bounce notification telling me the mail was not delivered and to check the Failed Deliveries page for details. That's working as I expected.

However, there's 2 things I noticed.
1. The notification was sent to a different alias target mail than the one it originated from (which happens to be my default alias target).
While understandable from a technical perspective, this seems counterintuitive.
Is this intended or a bug?
![image](https://user-images.githubusercontent.com/5447834/151605156-a74d97a5-713a-4fa6-ae2c-bb2a43eb9f4f.png)
2. The Failed Deliveries page shows ... nothing.
There's no mention of the mail bounced or any details as the mail notification hinted at.
This definitely feels like something is amiss here (either in container configuration or bounce handling)
![image](https://user-images.githubusercontent.com/5447834/151604868-492002bd-a55a-4f87-a5e7-8ae0a0f8d327.png)

## Setup details

- Running AnonAddy in Docker
- Release 0.8.10
- Currently still using opendkim/opendmarc milters

## Logs

Relevant logs I was able to extract (minus private details)
```
Jan 28 17:54:58 s01 postfix/smtpd[89318]: connect from localhost[127.0.0.1]
mail.info: Jan 28 16:54:58 opendmarc[844]: ignoring connection from localhost
Jan 28 17:54:58 s01 postfix/smtpd[89318]: 48B7A40921: client=localhost[127.0.0.1]
Jan 28 17:54:58 s01 postfix/cleanup[89323]: 48B7A40921: message-id=<83d727890e1a9a66bcac00ef9a0b16a3@example.com>
mail.info: Jan 28 16:54:58 opendkim[840]: 48B7A40921: DKIM-Signature field added (s=default, d=example.com)
Jan 28 17:54:58 s01 postfix/qmgr[1106]: 48B7A40921: from=<alias1@example.com>, size=14001, nrcpt=1 (queue active)
Jan 28 17:54:58 s01 postfix/smtpd[89318]: disconnect from localhost[127.0.0.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
Jan 28 17:54:58 s01 postfix/pipe[89324]: 51A3E408FC: to=<alias1+support=example.com@example.com>, relay=anonaddy, delay=1.5, delays=0.7/0.03/0/0.79, dsn=2.0.0, status=sent (delivered via anonaddy service)
Jan 28 17:54:58 s01 postfix/qmgr[1106]: 51A3E408FC: removed
Jan 28 17:54:58 s01 postfix/smtp[89334]: Trusted TLS connection established to XXX.protection.outlook.com[XXX.XXX.XXX.XXX]:25: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
Jan 28 17:54:58 s01 postfix/smtp[89334]: 48B7A40921: to=<support@example.com>, relay=XXX.protection.outlook.com[XXX.XXX.XXX.XXX]:25, delay=0.39, delays=0.03/0.02/0.23/0.12, dsn=5.7.606, status=bounced (host XXX.protection.outlook.com[XXX.XXX.XXX.XXX] said: 550 5.7.606 Access denied, banned sending IP [XXX.XXX.XXX.XXX]. To request removal from this list please visit https://sender.office.com/ and follow the directions. For more information please go to  http://go.microsoft.com/fwlink/?LinkID=526655 AS(1430) [XXX.prod.protection.outlook.com] (in reply to RCPT TO command))
Jan 28 17:54:58 s01 postfix/cleanup[89323]: AD0EE40930: message-id=<20220128165458.AD0EE40930@s01.example.com>
Jan 28 17:54:58 s01 postfix/bounce[89335]: 48B7A40921: sender non-delivery notification: AD0EE40930
Jan 28 17:54:58 s01 postfix/qmgr[1106]: AD0EE40930: from=<>, size=17194, nrcpt=1 (queue active)
Jan 28 17:54:58 s01 postfix/qmgr[1106]: 48B7A40921: removed

Jan 28 17:54:59 s01 postfix/smtpd[89318]: connect from localhost[127.0.0.1]
mail.info: Jan 28 16:54:59 opendmarc[844]: ignoring connection from localhost
Jan 28 17:54:59 s01 postfix/smtpd[89318]: 9FE9140921: client=localhost[127.0.0.1]
Jan 28 17:54:59 s01 postfix/cleanup[89323]: 9FE9140921: message-id=<3239c63eea980e647a46b539aa0460c1@swift.generated>
mail.info: Jan 28 16:54:59 opendkim[840]: 9FE9140921: DKIM-Signature field added (s=default, d=example.com)
Jan 28 17:54:59 s01 postfix/qmgr[1106]: 9FE9140921: from=<anonaddy@example.com>, size=5538, nrcpt=1 (queue active)
Jan 28 17:54:59 s01 postfix/smtpd[89318]: disconnect from localhost[127.0.0.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
Jan 28 17:54:59 s01 postfix/pipe[89324]: AD0EE40930: to=<alias1@example.com>, relay=anonaddy, delay=1.2, delays=0.01/0/0/1.1, dsn=4.3.0, status=deferred (An error has occurred, please try again later. )
crond: USER anonaddy pid 89372 cmd php /var/www/anonaddy/artisan schedule:run --no-ansi --no-interaction --quiet
Jan 28 17:55:00 s01 postfix/smtp[89334]: Trusted TLS connection established to mx.example.com[XXX.XXX.XXX.XXX]:25: TLSv1.3 with cipher TLS_AES_128_GCM_SHA256 (128/128 bits) key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256
Jan 28 17:55:03 s01 postfix/smtp[89334]: 9FE9140921: to=<realmail@example.com>, relay=mx.example.com[XXX.XXX.XXX.XXX]:25, delay=3.4, delays=0.13/0/1.5/1.8, dsn=2.0.0, status=sent (250 ok dirdel)
Jan 28 17:55:03 s01 postfix/qmgr[1106]: 9FE9140921: removed
```

**Comment by crazy-max** [comment]
@willbrowningme Yes logs are in `/data/storage/logs` in the container.

**Comment by croxxx** [comment]
I just tried sending a mail to a nonexistent address @outlook.com to test this.
Of course that did not do me the favor of causing the same issue. The bounce shows up as expected.
Meanwhile I'm whitelisted for that server, running the latest Docker release and switched to rspamd which makes reproducing this harder.

**However**, I poked around and found one entry in the logs for that day:
```
[2022-01-28 16:54:59] production.ERROR: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'code' at row 1 (SQL: insert into `failed_deliveries` (`recipient_id`, `alias_id`, `bounce_type`, `remote_mta`, `sender`, `email_type`, `status`, `code`, `attempted_at`, `user_id`, `id`, `updated_at`, `created_at`) values (?, c989f6c3-fbe5-47b8-afb0-7e04e548afb4, hard, xxxxxxxxx.mail.protection.outlook.com, ?, R, 5.7.606, smtp; 550 5.7.606 Access denied, banned sending IP     [xxx.xxx.xxx.xxx]. To request removal from this list please visit     https://sender.office.com/ and follow the directions. For more information     please go to  http://go.microsoft.com/fwlink/?LinkID=526655 AS(1430)     [VE1EURxxxxxxx.eop-EUR03.prod.protection.outlook.com], 2022-01-28 16:54:58, 5859a76c-db04-409d-ac50-a07aa49f8441, 889225da-bb33-45c6-bd4b-aad1c4b1325c, 2022-01-28 16:54:59, 2022-01-28 16:54:59)) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'code' at row 1 (SQL: insert into `failed_deliveries` (`recipient_id`, `alias_id`, `bounce_type`, `remote_mta`, `sender`, `email_type`, `status`, `code`, `attempted_at`, `user_id`, `id`, `updated_at`, `created_at`) values (?, c989f6c3-fbe5-47b8-afb0-7e04e548afb4, hard, xxxxxxxxx.mail.protection.outlook.com, ?, R, 5.7.606, smtp; 550 5.7.606 Access denied, banned sending IP     [xxx.xxx.xxx.xxx]. To request removal from this list please visit     https://sender.office.com/ and follow the directions. For more information     please go to  http://go.microsoft.com/fwlink/?LinkID=526655 AS(1430)     [VE1EURxxxxxxx.eop-EUR03.prod.protection.outlook.com], 2022-01-28 16:54:58, 5859a76c-db04-409d-ac50-a07aa49f8441, 889225da-bb33-45c6-bd4b-aad1c4b1325c, 2022-01-28 16:54:59, 2022-01-28 16:54:59)) at /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php:712)
[stacktrace]
#0 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(672): Illuminate\\Database\\Connection->runQueryCallback()
#1 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(502): Illuminate\\Database\\Connection->run()
#2 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(454): Illuminate\\Database\\Connection->statement()
#3 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2951): Illuminate\\Database\\Connection->insert()
#4 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(1655): Illuminate\\Database\\Query\\Builder->insert()
#5 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1164): Illuminate\\Database\\Eloquent\\Builder->__call()
#6 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(994): Illuminate\\Database\\Eloquent\\Model->performInsert()
#7 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php(296): Illuminate\\Database\\Eloquent\\Model->save()
#8 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Support/helpers.php(263): Illuminate\\Database\\Eloquent\\Relations\\HasOneOrMany->Illuminate\\Database\\Eloquent\\Relations\\{closure}()
#9 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php(297): tap()
#10 /var/www/anonaddy/app/Console/Commands/ReceiveEmail.php(374): Illuminate\\Database\\Eloquent\\Relations\\HasOneOrMany->create()
#11 /var/www/anonaddy/app/Console/Commands/ReceiveEmail.php(84): App\\Console\\Commands\\ReceiveEmail->handleBounce()
#12 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Console\\Commands\\ReceiveEmail->handle()
#13 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#14 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#15 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#16 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
#17 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call()
#18 /var/www/anonaddy/vendor/symfony/console/Command/Command.php(298): Illuminate\\Console\\Command->execute()
#19 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#20 /var/www/anonaddy/vendor/symfony/console/Application.php(1005): Illuminate\\Console\\Command->run()
#21 /var/www/anonaddy/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
#22 /var/www/anonaddy/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#23 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\\Component\\Console\\Application->run()
#24 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#25 /var/www/anonaddy/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#26 {main}

[previous exception] [object] (PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'code' at row 1 at /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php:501)
[stacktrace]
#0 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(501): PDOStatement->execute()
#1 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(705): Illuminate\\Database\\Connection->Illuminate\\Database\\{closure}()
#2 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(672): Illuminate\\Database\\Connection->runQueryCallback()
#3 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(502): Illuminate\\Database\\Connection->run()
#4 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Connection.php(454): Illuminate\\Database\\Connection->statement()
#5 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2951): Illuminate\\Database\\Connection->insert()
#6 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(1655): Illuminate\\Database\\Query\\Builder->insert()
#7 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1164): Illuminate\\Database\\Eloquent\\Builder->__call()
#8 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(994): Illuminate\\Database\\Eloquent\\Model->performInsert()
#9 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php(296): Illuminate\\Database\\Eloquent\\Model->save()
#10 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Support/helpers.php(263): Illuminate\\Database\\Eloquent\\Relations\\HasOneOrMany->Illuminate\\Database\\Eloquent\\Relations\\{closure}()
#11 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php(297): tap()
#12 /var/www/anonaddy/app/Console/Commands/ReceiveEmail.php(374): Illuminate\\Database\\Eloquent\\Relations\\HasOneOrMany->create()
#13 /var/www/anonaddy/app/Console/Commands/ReceiveEmail.php(84): App\\Console\\Commands\\ReceiveEmail->handleBounce()
#14 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Console\\Commands\\ReceiveEmail->handle()
#15 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#16 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#17 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#18 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
#19 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call()
#20 /var/www/anonaddy/vendor/symfony/console/Command/Command.php(298): Illuminate\\Console\\Command->execute()
#21 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#22 /var/www/anonaddy/vendor/symfony/console/Application.php(1005): Illuminate\\Console\\Command->run()
#23 /var/www/anonaddy/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
#24 /var/www/anonaddy/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#25 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\\Component\\Console\\Application->run()
#26 /var/www/anonaddy/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#27 /var/www/anonaddy/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#28 {main}
"} 
```
Given that, it seems that there is some check in place that requires server response codes to have a certain length - which this one failed. I'm not sure sending this long bounce message is conformant behaviour on the MTA side, but I think even a faulty bounce should be listed to the greatest extent possible.

**Comment by crazy-max** [comment]
> `String data, right truncated: 1406 Data too long for column 'code'`

Could be fixed with 0.11.0 @willbrowningme ?

**Comment by willbrowningme** [comment]
This should be fixed in [v0.11.1](https://github.com/anonaddy/anonaddy/releases/tag/v0.11.1), I've updated the column type from string to text.
